### PR TITLE
BUG: fix html float display

### PR DESCRIPTION
**Pull request by saldanhad**
Part of #59876
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added an entry in the latest `doc/source/whatsnew/v3.0.0.rst` file if fixing a bug or adding a new feature.

Fixes:
- Updated :meth:`DataFrame._repr_html_` to pass the current float format option from :func:`get_option("display.float_format")` to :class:`DataFrameFormatter`, ensuring that the HTML output respects the configured float formatting.

### CODE REVIEW

File: doc/source/whatsnew/v3.0.0.rst [hunk 1]
```diff
@@ -620,6 +620,8 @@ I/O
 ^^^
 - Bug in :class:`DataFrame` and :class:`Series` ``repr`` of :py:class:`collections.abc.Mapping`` elements. (:issue:`57915`)
 - Bug in :meth:`.DataFrame.to_json` when ``"index"`` was a value in the :attr:`DataFrame.column` and :attr:`Index.name` was ``None``. Now, this will fail with a ``ValueError`` (:issue:`58925`)
+- Bug in :meth:`DataFrame._repr_html_` pass :func: ``get_option("display.float_format")`` to :class:`DataFrameFormatter`, such that HTML output respects the configured float formatting. (:issue:`59876`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:623]:
```suggestion
```

File: doc/source/whatsnew/v3.0.0.rst [hunk 2]
```diff
@@ -620,6 +620,8 @@ I/O
...
+- Bug in :meth:`DataFrame._repr_html_` pass :func:`get_option("display.float_format")` to :class:`DataFrameFormatter`, such that HTML output respects the configured float formatting (:issue:`59876`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v3.0.0.rst:624]:
```suggestion
- Bug in :meth:`DataFrame._repr_html_` which ignored the ``"display.float_format"`` option (:issue:`59876`)
```

File: pandas/tests/io/formats/test_info_repr_html.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    @pytest.mark.parametrize(
+        "data, format_option, expected_values",
+        [
+            ({"A": [12345.6789]}, "{:12.3f}", "12345.679"),
+            ({"A": [None]}, "{:.3f}", "None"),
+            ({"A": [""]}, "{:.2f}", ""),
+            ({"A": [112345.6789]}, "{:6.3f}", "112345.679"),
+        ],
+    )  # test cases
+    def test_float_formatting_html_output(self, data, format_option, expected_values):
```
[the remark this answers is not in the document]
Comment (saldanhad) [reply pandas/tests/io/formats/test_info_repr_html.py:16]: sure

File: pandas/tests/io/formats/test_format.py [hunk 4]
```diff
@@ -368,6 +369,29 @@ def test_repr_min_rows(self):
...
+            ({"A": [None]}, "{:.3f}", "None"),
+            ({"A": [""]}, "{:.2f}", ""),
+            ({"A": [112345.6789]}, "{:6.3f}", "112345.679"),
+        ],
+    )
+    def test_repr_float_formatting_html_output(
+        self, data, format_option, expected_values
+    ):
+        if format_option is not None:
+            set_option("display.float_format", format_option.format)
```
Comment (mroeschke) [inline pandas/tests/io/formats/test_format.py:385]: Can you use the `option_context` context manager here instead?

Comment (saldanhad) [reply pandas/tests/io/formats/test_format.py:385]: sure, sorry about that, have a better idea of the workflow now.

File: pandas/tests/io/formats/test_format.py [hunk 5]
```diff
@@ -368,6 +368,23 @@ def test_repr_min_rows(self):
             assert ".." not in repr(df)
             assert ".." not in df._repr_html_()
 
+    @pytest.mark.parametrize(
+        "data, format_option, expected_values",
+        [
+            ({"A": [12345.6789]}, "{:12.3f}", "12345.679"),
```
Comment (mroeschke) [inline pandas/tests/io/formats/test_format.py:374]: Nit: Could you make `data` here the scalar values e.g. `12345.6789` and then in the test body construct `df = DataFrame({"A": [data]})`?

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @saldanhad

**Comment by saldanhad** [comment]
Thanks for your help in formatting the tests @mroeschke

**Pull request merged**